docs/source-2.0/additional-specs/rules-engine/standard-library.rst

```diff
@@ -38,6 +38,53 @@ parameter is equal to the value ``false``:
     }
 
 
+.. _rules-engine-standard-library-coalesce:
+
+``coalesce`` function
+=====================
+
+Summary
+    Evaluates arguments in order and returns the first non-empty result, otherwise returns the result of the last
+    argument.
+Argument types
+    * This function is variadic and requires two or more arguments, each of type ``T`` or ``option<T>``
+    * All arguments must have the same inner type ``T``
+Return type
+    * ``coalesce(T, T, ...)`` → ``T``
+    * ``coalesce(option<T>, T, ...)`` → ``T`` (if any argument is non-optional)
+    * ``coalesce(T, option<T>, ...)`` → ``T`` (if any argument is non-optional)
+    * ``coalesce(option<T>, option<T>, ...)`` → ``option<T>`` (if all arguments are optional)
+Since
+    1.1
+
+The ``coalesce`` function provides null-safe chaining by evaluating arguments in order and returning the first
+non-empty result. If all arguments evaluate to empty, it returns the result of the last argument. This is
```

Comment:
> If all arguments evaluate to empty, it returns the result of the last argument
I found this a little confusing - the last argument here would be empty right?

Reply:
Yeah. If the last argument is empty it returns empty, if it's not empty, it returns a not-empty value. When evaluating this condition, the implementation doesn't need to inspect the result of the last condition and can just return it (empty or not).